src/data/projectIdSingleton.ts

```diff
@@ -0,0 +1,114 @@
+import {
+  PineconeUnexpectedResponseError,
+  PineconeConfigurationError,
+  PineconeUnknownRequestFailure,
+  mapHttpStatusError,
+} from '../errors';
+import type { PineconeConfigurationOptions } from '../pinecone';
+import { buildUserAgent } from '../utils';
+
+// We only ever want to call whoami a maximum of once per API key, even if there
+// are multiple instantiations of the Index class. So we use a singleton here
+// to cache projectIds by apiKey.
+export const ProjectIdSingleton = (function () {
+  const projectIds = {}; // map of apiKey to projectId
+
+  const _fetchProjectId = async (
```

Comment:
The `@hidden` flag tells typedoc to ignore / not include something in the generated docs site. I'm not sure how it will handle this private singleton class. Will have to do some docs experimentation to see. 

This previously had the hidden annotation when it was a static method on the `Pinecone` class, but now that it's embedded in an [IIFE](https://www.javascripttutorial.net/javascript-immediately-invoked-function-expression-iife/) and the singleton is not even exported from the module I think there's no way for a user to access it. So I'm hopeful that typedoc will just omit it without any need for flag.